ams/server/src/main/java/com/netease/arctic/server/optimizing/OptimizerGroupMetrics.java

```diff
@@ -0,0 +1,192 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netease.arctic.server.optimizing;
+
+import static com.netease.arctic.ams.api.metrics.MetricDefine.defineGauge;
+import static com.netease.arctic.server.optimizing.OptimizingStatus.PENDING;
+import static com.netease.arctic.server.optimizing.OptimizingStatus.PLANNING;
+import static com.netease.arctic.server.optimizing.TaskRuntime.Status.ACKED;
+import static com.netease.arctic.server.optimizing.TaskRuntime.Status.PLANNED;
+import static com.netease.arctic.server.optimizing.TaskRuntime.Status.SCHEDULED;
+
+import com.netease.arctic.ams.api.metrics.Gauge;
+import com.netease.arctic.ams.api.metrics.Metric;
+import com.netease.arctic.ams.api.metrics.MetricDefine;
+import com.netease.arctic.ams.api.metrics.MetricKey;
+import com.netease.arctic.server.metrics.MetricRegistry;
+import com.netease.arctic.server.resource.OptimizerInstance;
+import org.apache.iceberg.relocated.com.google.common.annotations.VisibleForTesting;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class OptimizerGroupMetrics {
+
+  @VisibleForTesting protected static final String GROUP_TAG = "group";
+  public static final MetricDefine OPTIMIZER_GROUP_PENDING_TASKS =
+      defineGauge("optimizer_group_pending_tasks")
+          .withDescription("Number of pending tasks in optimizer group")
+          .withTags(GROUP_TAG)
+          .build();
+
+  public static final MetricDefine OPTIMIZER_GROUP_EXECUTING_TASKS =
+      defineGauge("optimizer_group_executing_tasks")
+          .withDescription("Number of executing tasks in optimizer group")
+          .withTags(GROUP_TAG)
+          .build();
+
+  public static final MetricDefine OPTIMIZER_GROUP_PLANING_TABLES =
+      defineGauge("optimizer_group_planing_tables")
+          .withDescription("Number of planing tables in optimizer group")
+          .withTags(GROUP_TAG)
+          .build();
+
+  public static final MetricDefine OPTIMIZER_GROUP_PENDING_TABLES =
+      defineGauge("optimizer_group_pending_tables")
+          .withDescription("Number of pending tables in optimizer group")
+          .withTags(GROUP_TAG)
+          .build();
+
+  public static final MetricDefine OPTIMIZER_GROUP_EXECUTING_TABLES =
+      defineGauge("optimizer_group_executing_tables")
+          .withDescription("Number of executing tables in optimizer group")
+          .withTags(GROUP_TAG)
+          .build();
+
+  public static final MetricDefine OPTIMIZER_GROUP_OPTIMIZERS =
+      defineGauge("optimizer_group_optimizers")
```

Comment:
May be `optimizer_group_optimizer_instances` is better?

Reply:
done